packages/fxa-auth-server/test/local/routes/session.js

```diff
@@ -1579,3 +1579,278 @@ describe('/session/verify/verify_push', () => {
+      authenticatorAssuranceLevel: 1,
+    };
+
+    db.account = sinon.spy(() => Promise.resolve(verifiedAccount));
+    // Mock account with TOTP available (AAL 2)
+    availableAuthStub.returns(
+      Promise.resolve(new Set(['pwd', 'email', 'otp']))
+    );
+
+    request = mocks.mockRequest({
+      log,
+      credentials: unverifiedSessionToken,
+      payload: {},
+    });
+
+    const response = await runTest(route, request);
+
+    assert.calledOnceWithExactly(db.account, 'foo');
+    assert.calledOnceWithExactly(availableAuthStub, db, verifiedAccount);
+
+    assert.deepEqual(response, {
+      accountStatus: 'verified',
+      sessionStatus: 'mustVerify',
+      verificationMethod: 'totp-2fa',
+    });
+  });
+
+  it('with verified account with 2FA enabled and AAL1 verified session, AAL upgrade required', async () => {
+    const verifiedAccount = {
+      uid: 'foo',
+      primaryEmail: {
+        email: 'foo@example.org',
+        isVerified: true,
+      },
+    };
+
+    const sessionToken = {
+      uid: 'foo',
+      email: 'foo@example.org',
+      emailVerified: true,
+      mustVerify: false,
+      tokenVerified: true,
+      authenticatorAssuranceLevel: 1,
+    };
+
+    db.account = sinon.spy(() => Promise.resolve(verifiedAccount));
+    availableAuthStub.returns(
+      Promise.resolve(new Set(['pwd', 'email', 'otp']))
+    );
+
+    request = mocks.mockRequest({
+      log,
+      credentials: sessionToken,
+      payload: {},
+    });
+
+    const response = await runTest(route, request);
+
+    assert.calledOnceWithExactly(db.account, 'foo');
+    assert.calledOnceWithExactly(availableAuthStub, db, verifiedAccount);
+
+    assert.deepEqual(response, {
+      accountStatus: 'verified',
+      sessionStatus: 'mustUpgrade',
+      verificationMethod: 'totp-2fa',
+    });
+  });
+
+  it('with verified account with 2FA enabled and AAL2 verified session', async () => {
+    const verifiedAccount = {
+      uid: 'foo',
+      primaryEmail: {
+        email: 'foo@example.org',
+        isVerified: true,
+      },
+    };
+
+    // Session with AAL 2 (TOTP verification)
+    const sessionToken = {
+      uid: 'foo',
+      email: 'foo@example.org',
+      emailVerified: true,
+      mustVerify: false,
+      tokenVerified: true,
+      authenticatorAssuranceLevel: 2,
+    };
+
+    db.account = sinon.spy(() => Promise.resolve(verifiedAccount));
+    // Mock account with TOTP available (AAL 2)
+    availableAuthStub.returns(
+      Promise.resolve(new Set(['pwd', 'email', 'otp']))
+    );
+
+    request = mocks.mockRequest({
+      log,
+      credentials: sessionToken,
+      payload: {},
+    });
+
+    const response = await runTest(route, request);
+
+    assert.calledOnceWithExactly(db.account, 'foo');
+    assert.calledOnceWithExactly(availableAuthStub, db, verifiedAccount);
+
+    assert.deepEqual(response, {
+      accountStatus: 'verified',
+      sessionStatus: 'verified',
+    });
+  });
+});
```

Comment:
Thanks for all the tests and comments in the route endpoint! Makes it clear what's happening.